packages/pages/src/components/hoursStatus/hoursStatus.tsx

```diff
@@ -0,0 +1,156 @@
+  } else {
+    const interval = params.futureInterval;
+    time += interval ? interval.getStartTime("en-US", params.timeOptions) : "";
+  }
+  return <span className="HoursStatus-time"> {time}</span>;
+}
+
+function defaultDayOfWeekTemplate(params: StatusParams): React.ReactNode {
+  const dayOptions: Intl.DateTimeFormatOptions = {
+    weekday: "long",
+    ...(params.dayOptions ?? {}),
+  };
+
+  let dayOfWeek = "";
+  if (params.isOpen) {
+    const interval = params.currentInterval;
+    dayOfWeek += interval?.end?.toLocaleString("en-US", dayOptions) || "";
+  } else {
+    const interval = params.futureInterval;
+    dayOfWeek += interval?.start?.toLocaleString("en-US", dayOptions) || "";
+  }
+  return <span className="HoursStatus-dayOfWeek"> {dayOfWeek}</span>;
+}
+
+function defaultStatusTemplate(
+  params: StatusTemplateParams,
+  props?: HoursStatusProps
+): React.ReactNode {
+  const currentTemplate = params.currentTemplate || defaultCurrentTemplate;
+  const separatorTemplate =
+    params.separatorTemplate || defaultSeparatorTemplate;
+  const futureTemplate = params.futureTemplate || defaultFutureTemplate;
+  const timeTemplate = params.timeTemplate || defaultTimeTemplate;
+  const dayOfWeekTemplate =
+    params.dayOfWeekTemplate || defaultDayOfWeekTemplate;
+
+  return (
+    <div className={c("HoursStatus", props?.className || "")}>
+      {currentTemplate(params)}
+      {separatorTemplate(params)}
+      {futureTemplate(params)}
+      {timeTemplate(params)}
+      {dayOfWeekTemplate(params)}
+    </div>
+  );
+}
+
+/*
+ * The HoursStatus component uses Hours data to generate a status message
+ *  describing the current Open/Closed status of the entity
+ *
+ * @param {HoursType} hours data from Yext Streams
+ * @param {Intl.DateTimeFormatOptions} timeOptions
+ * @param {Intl.DateTimeFormatOptions} dayOptions
+ * @param {Function} statusTemplate completely override rendering for this component
+ * @param {Function} currentTemplate override rendering for the "current" part of this component "[[Open Now]] - closes at 5:00PM Monday"
+ * @param {Function} separatorTemplate override rendering for the "separator" part of this component "Open Now [[-]] closes at 5:00PM Monday"
+ * @param {Function} futureTemplate override rendering for the "future" part of this component "Open Now - [[closes at]] 5:00PM Monday"
+ * @param {Function} timeTemplate override rendering for the "time" part of this component "Open Now - closes at [[5:00PM]] Monday"
+ * @param {Function} dayOfWeekTemplate override rendering for the "dayOfWeek" part of this component "Open Now - closes at 5:00PM [[Monday]]"
+ */
+const HoursStatus: React.FC<HoursStatusProps> = (props) => {
+  const [hasStatusTimeout, setHasStatusTimeout] = useState(false);
+
+  // Use two rendering passes to avoid SSR issues where server & client rendered content are different
+  // https://reactjs.org/docs/react-dom.html#hydrate
+  const [isClient, setIsClient] = useState(false);
```

Comment:
We have a util to check for client. Could that work here instead?

Reply:
Renamed variable to be a better representation of what it is doing